File: scripts/remove-th-vi.py
#!/usr/bin/env python3
# remove-th-vi.py - 删除 i18n.ts 中的 th: 和 vi: 翻译对象

import os

# 使用脚本所在目录来构造正确的文件路径
script_dir = os.path.dirname(os.path.abspath(__file__))
file_path = os.path.join(script_dir, '..', 'lib', 'i18n.ts')
file_path = os.path.normpath(file_path)

print(f'正在处理文件: {file_path}')

with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# 删除 th: { ... } 对象
# 使用非贪婪匹配到第一个独立的 "  }," 行
th_pattern = r'\n  th:\s*\{[\s\S]*?\n  \},\n'
content = re.sub(th_pattern, '\n', content)

# 删除 vi: { ... } 对象
# 匹配到 "  },\n};" 或 "  }\n};"
vi_pattern = r'\n  vi:\s*\{[\s\S]*?\n  \},\n\};\n'
content = re.sub(vi_pattern, '\n};\n', content)

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(content)

print('✅ 已删除 th 和 vi 翻译对象')

File: scripts/remove-th-vi.mjs
// 删除 i18n.ts 中的 th: 和 vi: 翻译对象
import { readFileSync, writeFileSync } from 'fs';

const filePath = new URL('../lib/i18n.ts', import.meta.url).pathname;
const content = readFileSync(filePath, 'utf-8');

// 删除 th: { ... } 对象（从 "  th: {" 到 "  }," 之前）
// 使用非贪婪匹配到第一个 "  },"（th对象的结束）
const thPattern = /\s+th:\s*\{[\s\S]*?\n  \},\n/;
const step1 = content.replace(thPattern, '\n');

// 删除 vi: { ... } 对象（从 "  vi: {" 到 "  }," 之前）
const viPattern = /\s+vi:\s*\{[\s\S]*?\n  \},\n?\}/;
const step2 = step1.replace(viPattern, '\n};

export');

// 修复可能的格式问题
const result = step2.replace(/export;\s*$/, 'export default translations;');

writeFileSync(filePath, result, 'utf-8');
console.log('✅ 已删除 th 和 vi 翻译对象');
